modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -169,7 +169,25 @@ GrainTracker::doesFeatureIntersectBoundary(unsigned int feature_id) const
   if (feature_index != invalid_size_t)
   {
     mooseAssert(feature_index < _feature_sets.size(), "Grain index out of bounds");
-    return _feature_sets[feature_index]._intersects_boundary;
+    return _feature_sets[feature_index]._boundary_intersection != BoundaryIntersection::NONE;
+  }
+
+  return false;
+}
+
+bool
+GrainTracker::isFeaturePercolated(unsigned int feature_id) const
+{
+  // TODO: This data structure may need to be turned into a Multimap
+  mooseAssert(feature_id < _feature_id_to_local_index.size(), "Grain ID out of bounds");
+
+  auto feature_index = _feature_id_to_local_index[feature_id];
+  if (feature_index != invalid_size_t)
+  {
+    mooseAssert(feature_index < _feature_sets.size(), "Grain index out of bounds");
+    return (_feature_sets[feature_index]._boundary_intersection &
+            (BoundaryIntersection::PRIMARY_PERCOLATION_BOUNDARY |
+             BoundaryIntersection::SECONDARY_PERCOLATION_BOUNDARY)) != BoundaryIntersection::NONE;
```

Comment:
This logic will also need to be fixed up. Yeah, I know it's duplicated, but the other part of this routine does differ from the parent class.

Reply:
Oh good catch